core/services/relay/evm/mercury/v2/data_source.go

```diff
@@ -25,7 +25,7 @@ import (
 )
 
 type Runner interface {
-	ExecuteRun(ctx context.Context, spec pipeline.Spec, vars pipeline.Vars, l logger.Logger) (run pipeline.Run, trrs pipeline.TaskRunResults, err error)
+	ExecuteRun(ctx context.Context, spec pipeline.Spec, vars pipeline.Vars, l logger.Logger) (run *pipeline.Run, trrs pipeline.TaskRunResults, err error)
```

Comment:
We were copying around these values even though we end up needing a pointer for later uses anyways. Using a pointer also fixes the linter warnings indirectly.